components/BottomSheet/src/MDCBottomSheetTransitionController.m

```diff
@@ -39,95 +42,88 @@ @implementation MDCBottomSheetTransitionController
     animationControllerForPresentedController:(__unused UIViewController *)presented
                          presentingController:(__unused UIViewController *)presenting
                              sourceController:(__unused UIViewController *)source {
-  return self;
+  return [[MDCBottomSheetTransition alloc] initWithPresenting:YES];
 }
 
 - (id<UIViewControllerAnimatedTransitioning>)
     animationControllerForDismissedController:(__unused UIViewController *)dismissed {
+  return [[MDCBottomSheetTransition alloc] initWithPresenting:NO];
+}
+
+@end
+
+@implementation MDCBottomSheetTransition {
+  BOOL _presenting;
+}
+
+- (instancetype)initWithPresenting:(BOOL)presenting {
+  self = [super init];
+  if (self) {
+    _presenting = presenting;
+  }
   return self;
 }
 
 #pragma mark - UIViewControllerAnimatedTransitioning
 
 - (NSTimeInterval)transitionDuration:
     (nullable __unused id <UIViewControllerContextTransitioning>)transitionContext {
-  return MDCBottomSheetTransitionDuration;
+  return MDCBottomSheetMotionSpec.transitionDuration;
 }
 
 - (void)animateTransition:(id <UIViewControllerContextTransitioning>)transitionContext {
-  // If a view in the transitionContext is nil, it likely hasn't been loaded by its ViewController
-  // yet.  Ask for it directly to initiate a loadView on the ViewController.
   UIViewController *fromViewController =
       [transitionContext viewControllerForKey:UITransitionContextFromViewControllerKey];
   UIView *fromView = [transitionContext viewForKey:UITransitionContextFromViewKey];
   if (fromView == nil) {
     fromView = fromViewController.view;
   }
+  if (fromView != nil && fromView == fromViewController.view) {
```

Comment:
The docs imply that it's possible for viewForKey: to return nil:

> viewForKey: may return nil which would indicate that the animator should not manipulate the associated view controller's view.

Reply:
> When will fromView != fromViewController.view?

I'm not sure when/if this could ever happen to be honest - it's possible there's some sort of UIKit edge case that the documentation is referring to?